manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/AbstractWorker.java

```diff
@@ -154,4 +180,29 @@ protected boolean isTimeoutExceeded(Work work, ManagedResource managedResource)
     protected abstract boolean isProvisioningComplete(T managedResource);
 
     protected abstract boolean isDeprovisioningComplete(T managedResource);
+
+    protected void recordAttemptAndReschedule(JobExecutionContext context) {
+        JobDataMap data = context.getTrigger().getJobDataMap();
+        long attempts = data.getLong(STATE_FIELD_ATTEMPTS);
+        data.put(STATE_FIELD_ATTEMPTS, String.valueOf(attempts + 1));
+
+        reschedule(context);
+    }
+
+    protected void reschedule(JobExecutionContext context) {
+        try {
+            Trigger existingTrigger = context.getTrigger();
+            TriggerKey existingTriggerKey = existingTrigger.getKey();
+            Trigger newTrigger = TriggerBuilder.newTrigger()
+                    .forJob(existingTrigger.getJobKey())
+                    .withIdentity(existingTriggerKey)
+                    .usingJobData(existingTrigger.getJobDataMap())
+                    .startAt(new Date(System.currentTimeMillis() + frequencySeconds * 1000))
+                    .build();
+            quartz.rescheduleJob(existingTriggerKey, newTrigger);
+        } catch (SchedulerException e) {
+            LOGGER.error(String.format("Unable to reschedule Job for '%s", context.getTrigger().getJobDataMap().getString(STATE_FIELD_ID)), e);
```

Comment:
@manstis We need to consider how to handle this failure because otherwise I think the associated resource (Bridge or Processor) simply ends-up stuck in the current `status`. Is my understanding correct?